app/src/stores/expenseStore.ts

```diff
@@ -0,0 +1,94 @@
+import { useCustomFetch } from '@/composables/useCustomFetch'
+import { useToastStore, useUserDataStore } from '@/stores'
+import type { ManyExpenseResponse } from '@/types'
+import { log } from '@/utils/generalUtil'
+import { defineStore } from 'pinia'
+import { computed, onMounted, ref, watch } from 'vue'
+import { useRoute } from 'vue-router'
+
+export const useExpenseDataStore = defineStore('expense', () => {
+  // TODO: fetch teams on mounted
+  // Having caching system,
+  // And trigger fetch on demand
+  // Update cache system.
+
+  const userDataStore = useUserDataStore()
+  const route = useRoute()
+  const { addErrorToast } = useToastStore()
+
+  const allExpenseURI = ref<string>(`/expense?teamId=${route.params.id}`)
+
+  const {
+    isFetching: allExpenseDataIsFetching,
+    error: allExpenseDataError,
+    data: allExpenseData,
+    execute: executeFetchAllExpenseData,
+    statusCode: allExpenseDataStatusCode
+  } = useCustomFetch(allExpenseURI, { immediate: false }).get().json()
+
+  const myApprovedExpenses = computed(() => {
+    if (allExpenseData.value) {
+      const expenses = allExpenseData.value.map(
+        (expense: { data: string; signature: `0x${string}` }) => ({
+          ...JSON.parse(expense.data),
+          signature: expense.signature
+        })
```

Comment:
Type directly the return of useCustomFetch so you don't have to type expense there.
It's a little bit tricky to understand here